### PR TITLE
Validation fails to continue when @app is nil

### PR DESCRIPTION
**Pull request by taquitos**
Checks to make sure @app is not nil, otherwise bails early and reports failure

### CODE REVIEW

File: app/services/boarding_service.rb [hunk 1]
```diff
@@ -74,9 +74,13 @@ def ensure_values
       spaceship = Spaceship::Tunes.login(@user, @password)
```
Comment (ohayon) [inline app/services/boarding_service.rb]: I think you could raise the error before this line if user, password, or app_id are empty right?

File: app/services/boarding_service.rb [hunk 2]
```diff
@@ -74,9 +74,13 @@ def ensure_values
...
       spaceship.select_team
 
-      @app ||= Spaceship::Tunes::Application.find(@app_id)
-      error_message << "Could not find app with ID #{app_id}" if @app == nil
-
+      @app ||= Spaceship::Tunes::Application.find(@app_id)      
+      if @app.nil?
```
Comment (ohayon) [inline app/services/boarding_service.rb]: then you could change this to something like `raise "Could not find app with ID #{app_id}"` if @app.nil?

File: app/services/boarding_service.rb (hunk 1 above)
Comment (taquitos) [inline app/services/boarding_service.rb]: 👍

File: app/services/boarding_service.rb (hunk 2 above)
Comment (taquitos) [inline app/services/boarding_service.rb]: 🌮

File: app/services/boarding_service.rb [hunk 3]
```diff
@@ -67,16 +67,16 @@ def add_tester(email, first_name, last_name)
...
+      raise error_message.join("\n") if error_message.length > 0
 
       spaceship = Spaceship::Tunes.login(@user, @password)
       spaceship.select_team
 
-      @app ||= Spaceship::Tunes::Application.find(@app_id)
-      error_message << "Could not find app with ID #{app_id}" if @app == nil
-
+      @app ||= Spaceship::Tunes::Application.find(@app_id)      
+      raise "Could not find app with ID #{app_id}" if @app.to_s.length == 0
```
Comment (ohayon) [inline app/services/boarding_service.rb]: Is there a reason why this isn't `if @app.nil?`?

Comment (taquitos) [inline app/services/boarding_service.rb]: Whoops, I was treating it like app_id, changing this back to `@app.nil?`

**Pull request merged**